ipfs-cluster-service/main.go

```diff
@@ -337,13 +350,34 @@ func setupDebug() {
 	//SetFacilityLogLevel("libp2p-raft", l)
 }
 
-func initConfig(force bool) {
+func initConfig(force bool, generateSecret bool, envSecret []byte) {
 	if _, err := os.Stat(configPath); err == nil && !force {
 		err := fmt.Errorf("%s exists. Try running with -f", configPath)
 		checkErr("", err)
 	}
+
 	cfg, err := ipfscluster.NewDefaultConfig()
 	checkErr("creating default configuration", err)
+
+	if !generateSecret {
+		if len(envSecret) != 0 {
+			// read cluster secret from env variable
+			fmt.Println("Reading cluster secret from CLUSTER_SECRET environment variable.")
+			cfg.ClusterSecret = []byte(envSecret)
+		} else {
+			// get cluster secret from user
+			isValid := false
+			for !isValid {
```

Comment:
You can directly abort (`checkErr()`) the first time the user screws up. Makes no differencein our case (we only have 1 question) and facilitates testing.